src/client/components/CampaignBadge/CampaignBadge.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react'
+import styled from '@emotion/styled'
+
+import { Badge, BadgeProps } from 'components/Badge/Badge'
+
+import { useAppliedCampaignNameQuery } from 'data/graphql'
+import { OfferData } from 'pages/OfferNew/types'
+import { isNorwegianBundle } from 'pages/OfferNew/utils'
+import { useTextKeys } from 'utils/textKeys'
+import { useCurrentLocale } from 'l10n/useCurrentLocale'
+
+const CampaignBadgeStyled = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: flex-start;
+  flex-wrap: wrap;
+  align-items: flex-start;
+
+  & > *:not(:last-child) {
+    margin-right: 0.5rem;
+  }
+`
+
+export type CampaignBadgeProps = Omit<BadgeProps, 'children'> & {
+  offerData: OfferData
+  quoteCartId: string
+}
+
+export const CampaignBadge: React.FC<CampaignBadgeProps> = ({
+  className,
+  quoteCartId,
+  offerData,
+  ...others
+}) => {
+  const { isoLocale } = useCurrentLocale()
+  const textKeys = useTextKeys()
+
+  const { data: campaignData, loading } = useAppliedCampaignNameQuery({
+    variables: {
+      id: quoteCartId,
+      locale: isoLocale,
+    },
+  })
+  const campaignText = campaignData?.quoteCart.campaign?.displayValue ?? ''
+
+  if (loading || !campaignText) return null
+
+  const discounts: Array<React.ReactNode> = [
+    ...(isNorwegianBundle(offerData)
```

Comment:
Can someone explain the use case for this check please? I'd love to be able to remove the dependency with `offerData` here.

Reply:
Hmmmm you could consider only passing the required data: `<CampaignBadge isNorwegianBundle={true/false}>` instead of `offerData`.

This is a static discount applied to all bundles in NO. However, this really should be something we apply automatically in Giraffe -- I think that's the solution going forward. So perhaps we keep this code until then?